server/notifications.go

```diff
@@ -0,0 +1,95 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package server
+
+import (
+	"context"
+	"sync"
+
+	"github.com/gofrs/uuid"
+)
+
+const (
+	// how many notifications can be in a notification buffer.
+	notificationBufferSize = 16
```

Comment:
Why 16? Why not just 1? 16 seems like a magic number here.

Reply:
The idea is that we should not normally exhaust these notification channels for good performance. 16 is a magic number, it could be 32 or 64. It's a balancing act between allocation and runtime performance. I just didn't think it was a good idea to expose this as a part of the config.